framework/src/modules/http_api/helpers/filter_peers.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+const _ = require('lodash');
+
+/**
+ * Sorts peers.
+ *
+ * @todo Add @param tags
+ * @todo Add @returns tag
+ * @todo Add description of the function
+ */
+
+const sortPeers = (field, asc) => (a, b) => {
+	// Match the default JavaScript sort order.
+	if (a[field] === b[field]) {
+		return 0;
+	}
+	// Ascending
+	if (asc) {
+		// Undefined last
+		if (a[field] === undefined) {
+			return 1;
+		}
+		if (b[field] === undefined) {
+			return -1;
+		}
+		// Null second last
+		if (a[field] === null) {
+			return 1;
+		}
+		if (b[field] === null) {
+			return -1;
+		}
+		if (a[field] < b[field]) {
+			return -1;
+		}
+
+		return 1;
+	}
+	// Descending
+	// Undefined first
+	if (a[field] === undefined) {
+		return -1;
+	}
+	if (b[field] === undefined) {
+		return 1;
+	}
+	// Null second
+	if (a[field] === null) {
+		return -1;
+	}
+	if (b[field] === null) {
+		return 1;
+	}
+	if (a[field] < b[field]) {
+		return 1;
+	}
+
+	return -1;
+};
+
+/**
+ * Returns peers length by filter but without offset and limit.
+ * @param {Array} peers
+ * @param {Object} filter
+ * @returns {int} count
+ * @todo Add description for the params
+ */
+const getByFilter = (peers, filter) => {
+	const allowedFields = [
+		'ip',
+		'wsPort',
+		'httpPort',
+		'state',
+		'os',
+		'version',
+		'protocolVersion',
+		'broadhash',
+		'height',
+		'nonce',
+	];
+	const limit = filter.limit ? Math.abs(filter.limit) : null;
+	const offset = filter.offset ? Math.abs(filter.offset) : 0;
+
+	peers = peers.filter(peer => {
```

Comment:
does this need to be reassigned?

Reply:
It needs to be reassigned because below, the filtered list is being used to sort and filter based on `limit` and `offset`. But for clarity, I have refactored this by assigning it to a var and using it further